### PR TITLE
Add lint checks and fix preexisting lints.

### PR DESCRIPTION
**Pull request by sengi**
Add [golangci-lint](https://golangci-lint.run/) as a pre-merge check, with GitHub review integration etc. and a make target for running it locally.

Fix preexisting lints; see commit messages.

Recommend reviewing commit-by-commit.

### CODE REVIEW

**Comment by sengi** [comment]
cc @chao-xian

**Comment by chao-xian** [comment]
wow this is a big PR...

**Comment by KludgeKML** [comment]
FWIW, it looks pretty reasonable. Given it's adding linting, all the changes look like the exact sort of thing linting would bring up, so if it's passing tests I feel pretty confident that it's a decent set of changes.

**Comment by sengi** [comment]
It's just a lot of mostly small commits that need to go in together.

I've kept roughly to fixing one type of issue per commit because it'll be much easier to review that way (especially since we have very few people familiar with Go).

**Comment by sengi** [comment]
Btw my overall aim here is to get the Router codebase into better shape so that the Postgres project has a better chance of going smoothly. This should make it a lot easier and safer to work on.

**Comment by chao-xian** [comment]
This mostly looks ok to me but I'm not a GoLang developer so I'd suggest getting another dev who has that level of familiarity to check as well

**Comment by sengi** [comment]
Lol, thought I'd fixed the last of the flaky tests in #366 but apparently not 🙃 Let's try that again.

**Comment by sengi** [comment]
Doesn't seem to be new flakiness (or if it is, it's quite a rare flake). So I think we're good.

**Comment by chao-xian** [comment]
@sengi can another Platform Engineer such as @dj-maisy or @theseanything review and approve?

File: integration_tests/router_support.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func reloadRoutes(optionalPort ...int) {
 	resp, err := http.Post(fmt.Sprintf("http://127.0.0.1:%d/reload", port), "", nil)
 	Expect(err).NotTo(HaveOccurred())
 	Expect(resp.StatusCode).To(Equal(202))
+	resp.Body.Close()
```
Comment (theseanything) [inline integration_tests/router_support.go:47]: good catch!

Comment (sengi) [reply integration_tests/router_support.go:47]: ah I can't take the credit, there's [a linter for this](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/bodyclose.go) :D

**Review by theseanything — APPROVED** [review]
Seems sensible to me!

**Pull request merged**